src/main/java/io/quarkus/search/app/SearchService.java

```diff
@@ -68,6 +69,21 @@ public SearchResult<SearchHit> search(@RestQuery @DefaultValue(QuarkusVersions.L
                                         .boost(50.0f)));
                     }
                 })
+                // * Highlighters are going to use spans-with-classes so that we will have more control over styling the visual on the search results screen.
+                // * The idea is that we want to get ~5 sentences (we use a sentence boundary scanner and set the fragment size to 0)
+                //   If we decide that we want to limit that and return only parts of a sentence (if those are really long ... we limit things by setting some positive value as a fragment size.
+                // * No match size is there to make sure that we are still going to get the text even if the field didn't have a match in it.
+                // * The title in the Guide entity is `Length.LONG` long, so we use that as a max value for no-match size, but hopefully nobody writes a title that long...
+                .highlighter(
+                        f -> f.unified().noMatchSize(Length.LONG).numberOfFragments(5).fragmentSize(0).orderByScore(true)
+                                .tag("<span class=\"highlightedContent\">", "</span>")
+                                .boundaryScanner().sentence().end())
+                // If we'd want to highlight a title in some way, e.g. make the text bolder, or add some background tint:
+                .highlighter("highlighter_title",
+                        f -> f.unified().numberOfFragments(1).tag("<span class=\"highlightedTitle\">", "</span>"))
+                // * If there's no match in the full content we don't want to return anything.
+                // * Also content is really huge, so we want to only get small parts of the sentences (we are setting a fragment size)
+                .highlighter("highlighter_content", f -> f.unified().noMatchSize(0).fragmentSize(150))
```

Comment:
Also, can you enable HTML escaping? I see a few XML tags included in the output, and this could get dangerous.